scripts/apps/authoring/authoring/ValidateCharacters.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import ng from 'core/services/ng';
+import {gettext} from 'core/utils';
+
+interface IProps {
+    item: string;
+    field: string;
+    onChange: (field: string, isError: boolean) => void;
+}
+
+export class ValidateCharacters extends React.PureComponent<IProps> {
+    render() {
+        const {item, field, onChange} = this.props;
+        const config = ng.get('$rootScope').config;
+        let invalidCharError = null;
+
+        if (item == null || field == null) {
+            return null;
+        }
+
+        if (config != null && config.disallowedCharacters != null) {
+            const disallowedCharacters = config.disallowedCharacters.split('');
+            const invalidCharString = disallowedCharacters.filter((char) => item.includes(char)).join(', ');
```

Comment:
I wonder why it should do the validation in the render of a component, shouldn't it rather be done in the authoring directive and here only render the error messages if any?

Reply:
I don't think we need to do that, it's just like we do for char-count or line count.